etcdserver/server_test.go

```diff
@@ -858,24 +858,36 @@ func TestSnapshot(t *testing.T) {
 	srv.kv = mvcc.New(be, &lease.FakeLessor{}, &srv.consistIndex)
 	srv.be = be
 
+	ch := make(chan struct{}, 2)
+
+	go func() {
```

Comment:
The `Clone` action is triggered by `snapshot`, so it must be concurrent with `snapshot` to avoid blocking, justifying the goroutine for `st.Wait(2). Likewise, `SaveNoCopy` expected by `st.Wait(2)` calls `SaveSnap`, so the `p.Wait(1)` must run concurrently to `st.Wait(2)`.